Automattic/wp-calypso
# Redux: Convert countries state slice to TypeScript

#### Proposed Changes

This PR attempts to add TypeScript types to the `countries` Redux reducer slice.

In so doing this corrects an existing type bug: previously it was assumed that every state slice under `state.countries` was an array, but the `woocommerce` data slice is an object. As a result of fixing this, the reducer default has been changed; previously when `state.countries.woocommerce` was first initialized, it was `[]` like the other `countries` slices; its value was only set to an object once data had been retrieved. Now it will default to `{}`. This bug was added in https://github.com/Automattic/wp-calypso/pull/59876

#### Testing Instructions

This should mainly only change types so if the tests pass then there shouldn't be anything to do.

However, one thing that is altered by this PR is replacing lodash `get()` in the `getCountries` selector with native JS optional chaining. To verify that this doesn't cause any problems, it's probably a good idea to test that a countries list still loads. One way to test this is to visit checkout and verify that the list of countries in the billing details/contact details step shows up as expected.
